### ISSUE TITLE
Generics throw cast exception if value is not in the cache: type 'Null' is not a subtype of type

### ISSUE THREAD

**Issue by ferrerogg** [ivoleitao/stash#24]
Using generics throw cast exception when getting a value that is not in the cache.

For example

```
import 'package:stash/stash_api.dart';
import 'package:stash_memory/stash_memory.dart';

class Task {
  final int id;
  final String title;
  final bool completed;

  Task({required this.id, required this.title, this.completed = false});

  @override
  String toString() {
    return 'Task $id, "$title" is ${completed ? "completed" : "not completed"}';
  }
}

void main() async {
  // Creates a store
  final store = newMemoryCacheStore();

  // Creates a cache with a capacity of 10 from the previously created store
  final cache = store.cache<Task>(name: 'cache1', maxEntries: 10, eventListenerMode: EventListenerMode.synchronous)
    ..on<CacheEntryCreatedEvent<Task>>().listen((event) => print('Key "${event.entry.key}" added to the cache'));

  // ***TRY TO GET A VALUE THAT IS NOT IN CACHE***
  final task = await cache.get('task1');
  print("task1 should be null: $task");
  // *******

  // Adds a task with key 'task1' to the cache
  await cache.put('task1', Task(id: 1, title: 'Run cache store example', completed: true));
  // Retrieves the value from the cache
  print(await cache.get('task1'));
}
```

Running the example throws this exception:

```
Unhandled exception:
type 'Null' is not a subtype of type 'Task' in type cast
#0      new Future.value (dart:async/future.dart:334:59)
#1      new GenericCache.<anonymous closure>
package:stash/…/cache/generic_cache.dart:105
#2      GenericCache.get.<anonymous closure>.<anonymous closure>
package:stash/…/cache/generic_cache.dart:328
<asynchronous suspension>
#3      main
example\cache\example.dart:25
<asynchronous suspension>
```

**Comment by ivoleitao** [comment]
Thank you for reporting this. It is indeed a bug, a stupid one actually :-(. I've already fix it and will provide a new build this weekend. I will also add a unit test to guarantee that this is covered

**Comment by ivoleitao** [comment]
Fixed on the new 4.0.1 version
